File: 算法/392. 判断子序列/IsSubsequence.go
package _92__判断子序列

func isSubsequence(s string, t string) bool {
	sLen := 0
	tLen := 0
	for {
		if sLen == len(s) || tLen == len(t) {
			break
		}
		if s[sLen] == t[tLen] {
			sLen++
			tLen++
		} else {
			tLen++
		}
	}
	return sLen == len(s)
}

File: 算法/392. 判断子序列/IsSubsequence.java
public class IsSubsequence {

    class Solution {
        public boolean isSubsequence(String s, String t) {
            if (s.length() > t.length()) {
                return false;
            }
            char[] childChar = s.toCharArray();
            char[] parentChar = t.toCharArray();
            int count = 0;
            int childStart = 0;
            int parentStart = 0;
            while (childStart < childChar.length && parentStart < parentChar.length) {
                if (childChar[childStart] == parentChar[parentStart]) {
                    childStart++;
                    parentStart++;
                    count++;
                } else {
                    parentStart++;
                }
            }
            return count == childChar.length;
        }
    }

}
